publication-commons/src/main/java/no/unit/nva/publication/service/impl/ResourceService.java

```diff
@@ -163,13 +165,16 @@ public Publication createPublicationWithPredefinedCreationDate(Publication input
         return insertResource(newResource);
     }
 
-    public Publication createPublicationFromImportedEntry(Publication inputData) {
+    public Publication createPublicationFromImportedEntry(Publication inputData, ImportSource importSource) {
         Resource newResource = Resource.fromPublication(inputData);
         newResource.setIdentifier(identifierSupplier.get());
         newResource.setPublishedDate(inputData.getPublishedDate());
         newResource.setCreatedDate(inputData.getCreatedDate());
         newResource.setModifiedDate(inputData.getModifiedDate());
         newResource.setStatus(PUBLISHED);
+        if (importSource != null) {
```

Comment:
Need this one until eventually CreatePublishedPublicationHandler is removed

Reply:
can we use `nonNull()`